tests/tt_eager/python_api_testing/unit_testing/misc/test_moreh_matmul.py

```diff
@@ -350,5 +365,55 @@ def test_moreh_matmul_wo_output(params, device):
 def test_moreh_matmul_enable_cache(params, device, use_program_cache):
     torch.manual_seed(3072)
     for i in range(2):
-        passing = moreh_matmul(params, True, device)
+        passing = moreh_matmul(params, True, None, device)
         assert passing
```

Comment:
Same comment as https://github.com/tenstorrent/tt-metal/pull/8966#discussion_r1622522504 about testing program cache.

Reply:
Dear @TT-BrianLiu ,
I added a test case to check the program cache. Would you please review it?